debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java

```diff
@@ -710,6 +713,24 @@ public static SnapshotQueryMode parse(String value, String defaultValue) {
             .withDescription("Enables transaction metadata extraction together with event counting")
             .withDefault(Boolean.FALSE);
 
+    public static final Field TRANSACTION_STRUCT_MAKER = Field.create("transaction.struct.maker")
```

Comment:
I don't think we need these two classes to be configurable by user a least now. It is fine if the connectors can override them but I'd keep it closed now.

Reply:
If we'd like to keep this configurable then it should be part of an SPI and only one config option hould be used instead of two. In that case it might be better to introduce a class that would be a factory for those SPI interface implementations. `io.debezium.connector.SourceInfoStructMaker` was intended more as a utility vehicle to enable `SourceInfo` migration for version 1.

Thinking loud and summoning @Naros. Maybe it is about time that we'd open `io.debezium.service.spi.ServiceRegistry` to be configurable from a connector confiuration so user could inject its own implementation?

Reply:
Yep, I could imagine this being done one of two ways @jpechane (maybe others):

The first way would be a very simple approach where the user supplies a comma-separated list of services that should be spun up into the registry, i.e.:
```
services=fully.qualified.service.a,fully.qualfiied.service.b
```
The other approach would be a more traditional configuration style like `converters` and `transforms` where the user supplies a `services` configuration that defines keys, and then we fetch the specific service and perhaps some optional configuration options via lookups.

In both cases, these services would extend a `ServiceContributor` contract that ultimately acts as a thin wrapper around the construction of the service, so that we could define things like this:

```java
public interface SomeFancyStructMakerService extends Service {
}

public class UserSuppliedFancyStructMakerService implements SomeFancyStructMakerService {
}
```

It's verbose, but it keeps a solid SoC between the contributor aspect and the actual implementation, allowing Debezium to initial the registry with a default/standard implementation and the user overriding that service with their own.  wdyt?

Reply:
I refactored to a single factory spi interface/class, so we only need one config option, which can be overridden in a connector ([example](https://github.com/debezium/debezium-connector-vitess/pull/188/files#diff-1b9011e9a90b9250c8885c36b838ac867f597b3726cf037ec1fa0aefb47d8221R430)).

The shared service registry changes sound useful, but probably would be outside of the scope of this PR. Once they are added, we can refactor transaction metadata to use the shared service registry.